core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -131,24 +135,32 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
 
     readonly actionPayload: Signal<ActionPayload> = computed(() => {
         const clientData = this.clientData();
-        const { containers, languageId, id, personaTag } = this.pageData();
-        const { contentletsId } = containers.find((container) => {
-            return (
-                container.identifier === clientData.container.identifier &&
-                container.uuid === clientData.container.uuid
-            );
-        });
+        const { container, contentlet } = clientData;
+        const relationship = getContentletRelationship(contentlet, container);
+        const { containers, languageId, id: pageId, personaTag, personalization } = this.pageData();
+        const { contentletsId } = containers.find(
+            ({ identifier, uuid }) => identifier === container.identifier && uuid === container.uuid
+        );
+        const treeOrder = contentletsId.findIndex((id) => id === contentlet?.identifier).toString();
+
+        const treeNode = {
+            ...relationship,
+            personalization,
+            treeOrder,
+            pageId
+        };
 
         return {
             ...clientData,
             language_id: languageId.toString(),
-            pageId: id,
+            pageId,
             pageContainers: containers,
             personaTag,
             container: {
-                ...clientData.container,
+                ...container,
                 contentletsId
-            }
+            },
+            treeNode
```

Comment:
Should we normalize this data? there are some duplications here.

Reply:
Done!